### PR TITLE
Relicense from `Zlib` to `Zlib OR Apache-2.0 OR MIT`

### PR DESCRIPTION
**Pull request by Lokathor**
While it's quite clear that the Zlib license is strictly better for the user than the MIT license, it's also the case that Rust projects generally expect to have `Apache-2.0 OR MIT` as the license.

As `tinyvec` picks up in popularity, it's somehow become a dependency of the `rust-analyzer` project, and they're requesting that we add those licenses as options (https://github.com/Lokathor/tinyvec/issues/79).

This PR is to change the crate license from just `Zlib` to instead be `Zlib OR Apache-2.0 or MIT`.

The general understanding for what a re-license entails is that we need to have a sign off from all previous contributors or replace any contributions they made. In the extreme case of not being able to contact someone we could see about just replacing their contributions anyway, but I don't think that anyone has suddenly disappeared so we should be able to get a response from everyone.

Obviously as the person starting the PR, I'm agreeing to do this, but everyone else is about to get a github ping.

Please comment that you either agree or disagree with the license change. I'll check the boxes as people reply.
* [x] @Shnatsel made some contributions to the docs in the past, but they explained (on Discord) that their employer wouldn't allow a relicense without contacting the company's legal team and going through a whole thing, so I've simply replaced the crate docs from scratch, since they needed a bit of an update anyway.
* [x] @ThatsNoMoon 
* [x] @cuviper
* [x] @dhardy
* [x] @dxenonb
* [x] @rozaliev
* [x] @mental32
* [x] @mgostIH
* [x] @repnop
* [x] @HeroicKatora
* [x] @Nemo157

And while I've got your attention I'd like to again **thank everyone** who has contributed. `tinyvec` actually has the most contributors out of all the projects I've worked on and it's always been very nice getting contributions.

Closes https://github.com/Lokathor/tinyvec/issues/79

### CODE REVIEW

**Comment by HeroicKatora** [comment]
I license past and future contributions under `Zlib OR Apache-2.0 or MIT` license, allowing licensees to chose any one at their option.

**Comment by repnop** [comment]
Yup, no problems here. I license past and future contributions under Zlib OR Apache-2.0 OR MIT license, allowing licensees to chose any one at their option.

**Comment by Nemo157** [comment]
I feel my non-dev-only changes aren't significant enough to be copyrightable, but 🤷

I license past and future contributions under `Zlib OR Apache-2.0 or MIT` license, allowing licensees to chose any one at their option.

**Comment by rozaliev** [comment]
I license past and future contributions under `Zlib OR Apache-2.0 or MIT` license, allowing licensees to chose any one at their option.

**Comment by mental32** [comment]
Tldr +1

Slightly embarrassing as I can't remember what I've contributed but I'm happy to sign off on anything needed 🙂

**Comment by cuviper** [comment]
I license past and future contributions under Zlib OR Apache-2.0 or MIT license, allowing licensees to chose any one at their option.

**Comment by dxenonb** [comment]
I think my change was just docs also. Nonetheless, I also license past and future contributions under Zlib or Apache-2.0 or MIT license, allowing licensees to chose any one at their option.

**Comment by ThatsNoMoon** [comment]
I license past and future contributions under Zlib OR Apache-2.0 or MIT license, allowing licensees to chose any one at their option.

**Comment by dhardy** [comment]
I license past and future contributions under Zlib OR Apache-2.0 or MIT license, allowing licensees to chose any one at their option.

**Comment by mgostIH** [comment]
Sure thing, I license past and future contributions under Zlib OR Apache-2.0 or MIT license, allowing licensees to chose any one at their option.

**Comment by Lokathor** [comment]
That's everyone. Thanks all.
